### PR TITLE
Fix status code response

### LINKED ISSUE #122: Unexpected status code returned for non-existent collections in the database
### I'm submitting a 
- [x] bug report.

### Current Behaviour:

Unexpected status code returned by the server:
Currently the hydrus server returns a `status code 405` for non-existent collections in the database.
screenshot from 2018-02-25 00-19-08

### Expected Behaviour:

The server shall give a response with `status code 404` for these endpoints.

### Steps to reproduce:

To fix this behaviour we can add an `additional return value`  in the `checkEndpoint` function
[checkEndpoint](https://github.com/HTTP-APIs/hydrus/blob/master/hydrus/app.py/#L47)
which denotes the `expected status code` to be returned and abort that request with that status code.
### Do you want to work on this issue?

yes

### PR DESCRIPTION
**Pull request by alohamora**
Fixes #122 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

The following commits fix the behaviour of status codes returned by the hydrus server for non existing collections.
### Test log
NOTE - I didn't knew how to log tests, so posting a picture of the test runs. Please guide me on that.
screenshot from 2018-02-25 05-02-43

### Changed behaviour
The server initially returned status code 405 for any random url of form `/API_NAME/randomwords`.
The new `check_endpoint` function now provides a `status code 404` for collections not existing in the database. New response attatched below:
screenshot from 2018-02-25 05-11-19

### CODE REVIEW

**Comment by alohamora** [comment]
@xadahiya @chrizandr @Mec-iS please review the request and suggest any changes required.

**Comment by xadahiya** [comment]
@alohamora  please resolve the conflicts.

**Comment by alohamora** [comment]
@xadahiya synced with develop and added type annotations.

**Comment by vddesai1998** [comment]
Please run mypy static type analyzer if you have added type annotations. And post the error log if returned

**Comment by alohamora** [comment]
Ran mypy test analyser...there were no errors...Also I have used `Dict[str, Any]` for annotating the dictionary in `checkEndpoint`function...is that a precise notation or should I use another format. @vddesai1998

File: hydrus/app.py [hunk 1]
```diff
@@ -45,14 +45,20 @@ def hydrafy(object_: Dict[str, Any]) -> Dict[str, Any]:
     return object_
 
 
-def checkEndpoint(method: str, type_: str) -> bool:
+def checkEndpoint(method: str, type_: str) -> Dict[str, Any]:
```
Comment (vddesai1998) [inline hydrus/app.py]: You should use Union[bool, Dict[str, Any]] as return type

**Comment by vddesai1998** [comment]
Refer 
 https://mypy.readthedocs.io/en/latest/basics.html  for more guidance

**Comment by alohamora** [comment]
@xadahiya commited all the requested changes...please review the PR.

**Pull request merged**